lib/payment.js

```diff
@@ -204,8 +204,15 @@ class Payment {
             _signature.set(p, json.seals.wallet.signature);
         }
         if (json.sender && json.sender.data && typeof json.sender.data === 'string') {
-            const senderData = JSON.parse(Buffer.from(json.sender.data, 'base64').toString('ascii'));
-            _senderRef.set(p, senderData.ref);
+            const jsonSenderData = Buffer.from(json.sender.data, 'base64').toString('ascii');
+            try {
+                const senderData = JSON.parse(jsonSenderData);
+                _senderRef.set(p, senderData.ref);
+            }
+            catch (e) {
+                // Ignore parse errors and leave the _senderRef undefined.
+                // TODO: Is there an exception safe alternative to JSON.parse?
```

Comment:
The try/catch here seems fine to me. It not only validates the `JSON.parse` function but also the data structure `senderData.ref`. I am thinking if we don't throw the exception, the clients may not be aware of the potential issue in the data string that they presume fine.

Reply:
True, but the principle of the from() method is that it should *never* throw. If there is an error we should return null instead. In this case however the reason why the sender reference is not available is not really important.. it either is, or it is not there.